model/renderNativeWindow.h

```diff
@@ -24,12 +24,44 @@
 
 namespace minsky
 {
-  class RenderNativeWindow: public ecolab::CairoSurface
+  struct WindowInformation
+  {
+    unsigned long parentWindowId;
+    unsigned long childWindowId;
+    int childWidth;
+    int childHeight;
+    int offsetLeft;
+    int offsetTop;
+    Display*	display;
+    XWindowAttributes wAttr;
+    ecolab::cairo::SurfacePtr *childSurface; // TODO:: Review right way to store surface ptr
```

Comment:
A SurfacePtr is a smart pointer - basically a std::shared_ptr IIRC. Taking a pointer to it will almost invariably lead to resource leaks and object lifetime problems.